cpp/refactor_module/algorithm/refactor.cpp

```diff
@@ -517,6 +558,114 @@ void NormalizeToBoolean(NodeOrRel object, const std::string &property_key,
 
 }  // namespace
 
+void Refactor::DeleteAndReconnect(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::MemoryDispatcherGuard guard{memory};
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+  try {
+    const auto path{arguments[0].ValuePath()};
+    const auto nodes_to_delete_list{arguments[1].ValueList()};
+    const auto config_map{arguments[2].ValueMap()};
+
+    std::unordered_set<int64_t> nodes_to_delete;
+    for (const auto &node : nodes_to_delete_list) {
+      nodes_to_delete.insert(node.ValueNode().Id().AsInt());
+    }
+
+    Config config{config_map};
+    mgp::List nodes;
+    mgp::List relationships;
+    int64_t prev_non_deleted_path_index{-1};
+    int64_t prev_non_deleted_node_id{-1};
+    std::unordered_set<mgp::Node> to_be_deleted;
+    auto graph = mgp::Graph{memgraph_graph};
+
+    for (size_t i = 0; i < path.Length() + 1; ++i) {
+      auto node = path.GetNodeAt(i);
+      int64_t id = node.Id().AsInt();
+      auto delete_node = nodes_to_delete.contains(id);
+
+      auto modify_relationship = [&graph, &relationships](mgp::Relationship relationship, const mgp::Node node,
```

Comment:
This lambda can be `const auto`

Reply:
I think you can pass relationship and node by `&`

Reply:
Not possible to pass relationship by reference because sometimes temporary object gets sent as an argument, it is possible to some const_cast trickery but I saw that that is not very good practice